crates/bevy_ecs/src/query/iter.rs

```diff
@@ -153,23 +155,20 @@ where
     }
 }
 
-pub struct QueryCombinationIter<'w, 's, Q: WorldQuery, QF, F: WorldQuery, const K: usize>
+pub struct QueryCombinationIter<'w, 's, Q: WorldQuery, F: WorldQuery, const K: usize>
```

Comment:
Good catch dropping the unnecessary QF here since it is fixed to `Fetch: ReadOnlyQuery` at the entry point anyway.